airflow/models/param.py

```diff
@@ -121,9 +122,21 @@ def __init__(self, dict_obj: Optional[Dict] = None, suppress_exception: bool = F
                 params_dict[k] = Param(v)
             else:
                 params_dict[k] = v
-        super().__init__(params_dict)
+        self.__dict = params_dict
         self.suppress_exception = suppress_exception
 
+    def __contains__(self, o: object) -> bool:
+        return o in self.__dict
+
+    def __len__(self) -> int:
+        return len(self.__dict)
+
+    def __delitem__(self, v: str) -> None:
+        del self.__dict[v]
+
+    def __iter__(self):
+        return self.__dict.__iter__()
```

Comment:
```suggestion
        return iter(self.__dict)
```

For consistency—we didn’t use the dict’s magic methods for all other magic method implementations.